pallets/loans-ref/src/lib.rs

```diff
@@ -494,12 +494,13 @@ pub mod pallet {
 			pool_id: PoolIdOf<T>,
 			loan_id: T::LoanId,
 			amount: T::Balance,
+			unchecked_amount: T::Balance,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
 
 			let (amount, _count) = Self::update_active_loan(pool_id, loan_id, |loan| {
 				Self::ensure_loan_borrower(&who, loan.borrower())?;
-				loan.repay(amount)
+				loan.repay(amount, unchecked_amount)
 			})?;
```

Comment:
We should add `unchecked_amount: T::Balance` as the last parameter to the `Repaid` event.

Reply:
Makes sense!

Reply:
Actually, I see what's happening now, the `amount` logged in the event will include the unchecked amount. I fear an indexing layer needs to be able to know what the checked vs unchecked amount was.

Reply:
Maybe `Loan::repay` can return separate amount and unchecked amounts, rather than adding them together?

Reply:
Yes, In fact, I think `Loan::repay` shouldn't do anything with `unchecked` except stores it.